lib/ansible/modules/network/eos/eos_user.py

```diff
@@ -340,6 +341,12 @@ def main():
                            supports_check_mode=True)
 
     warnings = list()
+    if module.params['password'] and not module.params['configured_password']:
```

Comment:
https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec

Maybe `required_if` would do the same thing?